src/BinarySerializationTestForStructs.zig

```diff
@@ -0,0 +1,135 @@
+const std = @import("std");
+const builtin = @import("builtin");
+const file = std.fs.File;
+const cwd = std.fs.cwd();
+const fba = std.heap.FixedBufferAllocator;
+
+const ArbitraryType = struct {
+    size: u64,
```

Comment:
This seems like experimenting code, but you \*probably* want `usize` instead of `u64`, see https://ziglang.org/documentation/master/#toc-Primitive-Types.